Naveego Code Challenge Plugin Solution in Node

roehlerw opened the pull request:
Passes all tests except for one bonus test related to datetime. I compared the expected and my actual and it looked like they were the same so I am not sure why that fails.

Needs to have robust error handling added.

roehlerw committed:
first solution

roehlerw committed:
stage log file

roehlerw committed:
fix formating error

roehlerw committed:
update log

roehlerw committed:
add readme

roehlerw committed:
update readme

SteveRuble commented:
I've updated the host to compare dates correctly. I also added a more gnarly data file with only ten rows. For some reason your plugin publishes 21 rows from that file.

roehlerw committed:
Merge remote-tracking branch 'upstream/master'

roehlerw committed:
updated plugin to better handle garbage input

roehlerw commented:
Added better schema inferring logic. It now looks at all records and picks the most common one for each column. 
Added a check in publishing where it will not publish blank rows (caused garbage to publish 21 instead of 10 rows). 
All tests pass except for garbage which really would benefit from allowing a user to input what each column's data type is supposed to be.

roehlerw committed:
Update README.md
